src/Compilers/Test/Core/Platform/CoreClr/CoreCLRRuntimeEnvironment.cs

```diff
@@ -104,11 +110,201 @@ public int Execute(string moduleName, string[] args, string expectedOutput)
         public SortedSet<string> GetMemberSignaturesFromMetadata(string fullyQualifiedTypeName, string memberName) =>
             GetEmitData().GetMemberSignaturesFromMetadata(fullyQualifiedTypeName, memberName);
 
+        private class Resolver : ILVerify.ResolverBase
+        {
+            private readonly Dictionary<string, ImmutableArray<byte>> imagesByName = new Dictionary<string, ImmutableArray<byte>>();
+
+            internal Resolver(EmitData emitData)
+            {
+                foreach (var module in emitData.AllModuleData)
+                {
+                    string name = module.FullName;
+                    var image = module.Image;
+
+                    // TODO2 figure out why we need both the simple name and full name
```

Comment:
Is it possible that you need simple names when dealing with non-strong named compilations and full names when dealing with strong named compilations?